build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchBuildCompletePlugin.java

```diff
@@ -142,10 +142,16 @@ public void execute(BuildFinishedFlowAction.Parameters parameters) throws FileNo
             uploadFile.getParentFile().mkdirs();
             createBuildArchiveTar(parameters.getFilteredFiles().get(), parameters.getProjectDir().get(), uploadFile);
             if (uploadFile.exists() && "true".equals(System.getenv("BUILDKITE"))) {
-                String uploadFilePath = "build/" + uploadFile.getName();
+                String uploadFilePath = uploadFile.getName();
                 try {
                     System.out.println("Uploading buildkite artifact: " + uploadFilePath + "...");
-                    new ProcessBuilder("buildkite-agent", "artifact", "upload", uploadFilePath).start().waitFor();
+                    ProcessBuilder pb = new ProcessBuilder("buildkite-agent", "artifact", "upload", uploadFilePath);
+                    // If we don't switch to the build directory first, the uploaded file will have a `build/` prefix
+                    // Buildkite will flip the `/` to a `\` at upload time on Windows, which will make the search command below fail
+                    // So, if you change this such that the artifact will have a slash/directory in it, you'll need to update the logic
+                    // below as well
+                    pb.directory(new File("./build"));
```

Comment:
Using relative paths with `new File()` is predictable. Can't we just use `uploadFile.getParentFile()` here?

Reply:
Yep, that's better, thanks